Registration form with sections

totallynotmanas opened the pull request:
have put comment on the variable to edit IsInternal var

ngpal committed:
bare bones

totallynotmanas committed:
add a progress bar, and sectioning

totallynotmanas committed:
form validation for steps

totallynotmanas committed:
css fix

totallynotmanas committed:
css fix #2

adithya-menon-r reviewed (changes requested):
A few UI changes:
- The dropdown UI has to be fixed

![image](https://github.com/user-attachments/assets/f87180c5-dcf1-4205-b442-5007703e1b95)

- Also in preferences, for committees and countries please make it dropdowns instead of text boxes, use dummy data like a `json` for now. But make it dynamic, so if a committee was selected in a previous dropdown, it can't be chosen in the next. 

- Make the university field in external a dropdown too, but if their college isn't there, allow them to type it too, but as they type you can filter teh option in the dropdown and show, if nothing matches, use their input directly

- Add validation for phone number (10 digits), and pincode (6 digits) and show errors like in sign up page

- Add the field names outside instead of as the placeholder, makes it easier to review

totallynotmanas committed:
ui fix and changes for better visibility

added labels for text inputs
fixed visibility issue for dropdowns
and fixed css issue

adithya-menon-r committed:
Created a simple dashboard UI & integrated Supabase to fetch user data

adithya-menon-r committed:
Integrated supabase to fetchUser data in /register

ngpal committed:
smaller code

adithya-menon-r committed:
Reverted code deletion in 51d2f17

adithya-menon-r committed:
Fixed disabled properties bug

adithya-menon-r committed:
Updated progress bar UI

adithya-menon-r committed:
Made updates to general UI

adithya-menon-r reviewed (changes requested):
@ngpal @totallynotmanas So I went through and made a few updates. Please fix the following:

Preferences Section Bug:

https://github.com/user-attachments/assets/469ed7e3-47b6-45ab-ab3d-6c2882c4b91d

The previously filled data resets? after changing to IP/Delegate?

adithya-menon-r committed:
Fixed file selection bug

adithya-menon-r committed:
Made all personal details required

adithya-menon-r committed:
Made university name a dropdown field

adithya-menon-r committed:
Fixed college dropdown bug

adithya-menon-r committed:
Made payment fields required

adithya-menon-r committed:
Fixed all bugs in preferences section

adithya-menon-r committed:
Fetches actual data for country & committee field dropdowns

adithya-menon-r committed:
Integrated supabase for form submission

adithya-menon-r committed:
Made submission atomic & moved prefs data fetching to page.tsx

adithya-menon-r committed:
Fixed issues in transaction rollback code

adithya-menon-r committed:
Fixed final issues in prefs validation & logic

adithya-menon-r committed:
Added flags/logos to options in country/member prefs

adithya-menon-r reviewed (approved):
All issues fixed.
